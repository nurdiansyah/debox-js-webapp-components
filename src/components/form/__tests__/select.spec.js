import React from 'react'

import {Form, Select, getFormBlock, getSelectBlock} from '../index'

const values = [
  {label: 'Mangga', value: 'mangga'},
  {label: 'Mangga', value: 'mangga'},
  {id: '2', label: 'Jeruk', value: 'jeruk'},
  {id: '3', label: 'Apel', value: 'apel', checked: true}
]

describe('form/Select', () => {
  it('should running no error', () => {
    const formBlock = getFormBlock({})
    const selectBlock = getSelectBlock({})
    const wrapper = render(
      <Form block={formBlock}>
        <Select block={selectBlock} name="buah" values={values} />
      </Form>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
