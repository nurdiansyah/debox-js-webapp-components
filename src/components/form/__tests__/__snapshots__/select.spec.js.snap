// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form/Select should running no error 1`] = `
<form
  class="form"
>
  <div
    class="form__control"
  >
    <select
      class="select"
      id="buah1"
    >
      <option
        value="mangga"
      >
        Mangga
      </option>
      <option
        value="mangga"
      >
        Mangga
      </option>
      <option
        id="2"
        value="jeruk"
      >
        Jeruk
      </option>
      <option
        checked=""
        id="3"
        value="apel"
      >
        Apel
      </option>
    </select>
  </div>
</form>
`;
